src/index.js

```diff
@@ -124,83 +169,62 @@ const getAttribute = (attributes, name) => {
   return getAttributeValues(attributes, path)[0];
 };
 
-const getCompareValue = (condition, attributes) => {
-  if ('target' in condition) {
-    return getAttribute(attributes, condition.target);
-  }
-  return condition.value;
-};
-
 /**
- * @returns `true` if the comparision matches, `false` if there is a mismatch,
- *          and `undefined` if the target value is not known to compute the
- *          result
+ * @returns `true` if the comparision matches, `false` if there is a mismatch.
  */
-const compare = (condition, value, attributes) => {
-  const compareValue = getCompareValue(condition, attributes);
+const compare = (condition, value) => {
+  const compareValue = condition.value;
+
   switch (condition.comparison) {
     case 'includes':
-      if (compareValue === undefined) return undefined;
```

Comment:
@aecorredor I'm not seeing where this `undefined` scenario is handled elsewhere -- could you link me?

Reply:
@swain it gets handled by the reversion logic in `maybeReverseCondition`. Any time the target is unknown we're reversing it to the known value, so by the time `compare` gets called, there's already an actual value. Does that make sense? The test coverage actually helped identify this and did make the code simpler (compare doesn't need `attributes` now).

Reply:
Makes sense, and you're right that the code is simpler. Two follow-up ideas, both motivated by the fact that this code is sadly still JavaScript (not TypeScript):
1. Could you add some `@param` JSDoc entries to describe the expected shape of `condition` and `value`?

2. Should we `throw` if `compareValue` is `undefined`, just for sanity?

Reply:
Sounds good.

Regarding point 2: we could, but given how the code is currently executed, we'll probably have to ignore the coverage for that line, since it won't get hit. That's where TS would come into play to just tell you it's not a valid case. Does it still make sense to add it? I felt confident not to do it since we had 100% test coverage before and all tests are still passing.

Reply:
I'm convinced -- no need for the throw.